src/parser/shared/metrics/apl/conditions/repeatableBuffPresent.tsx

```diff
@@ -0,0 +1,72 @@
+import Spell from 'common/SPELLS/Spell';
+import { formatRange, Range } from './util';
+import { Condition, tenseAlt } from '../index';
+import { EventType } from 'parser/core/Events';
+import SpellLink from 'interface/SpellLink';
+
+/**
+ * Buffs that can occur as multiple unique instances, but do not stack. Examples include the buffs granted by each Elemental Spirit from Enhancement Shaman
+ * @param spell {@link Spell} or array of spells the condition should match
+ * @param range Range of occurances to match
+ * @param latencyOffset
+ */
+export function repeatableBuffPresent(
+  spell: Spell | Spell[],
+  range: Range,
+  latencyOffset: number = 0,
+): Condition<{ [spellId: number]: number }> {
+  if (!Array.isArray(spell)) {
+    spell = [spell];
+  }
+  const spellId = spell.map((s) => s.id);
+  return {
+    key: `repeatableBuffPresent-${spellId.join('-')}`,
+    init: () => [],
+    update: (state, event) => {
+      switch (event.type) {
+        case EventType.ApplyBuff:
+          if (spellId.includes(event.ability.guid)) {
+            state[event.ability.guid] ??= 0;
+            state[event.ability.guid] += 1;
+          }
+          break;
+        case EventType.RemoveBuff:
+          if (spellId.includes(event.ability.guid)) {
+            state[event.ability.guid] ??= 0;
+            state[event.ability.guid] -= 1;
+            if (state[event.ability.guid] === 0) {
+              delete state[event.ability.guid];
+            }
+          }
+          break;
+      }
+      return state;
+    },
+    validate: (state, _event) => {
+      const count = Object.keys(state).reduce((total, id) => (total += state[Number(id)]), 0);
+      return count >= (range.atLeast || 0) && (range.atMost === undefined || count <= range.atMost);
+    },
+    describe: (tense) => (
+      <>
+        you {tenseAlt(tense, 'have', 'had')} {formatRange(range)} of{' '}
+        <>
+          {spell.length === 1 ? (
+            <SpellLink spell={spell[0]} />
+          ) : (
+            <>
+              <ul>
```

Comment:
not sure how i feel about `<ul>` here but this is probably a fine default. i feel like these are generally cases where you want a shorthand for it (like "Elemental Spirit buffs") rather than a list of all the options.

Reply:
This was more a solution due to my lack of react skills. I'd much rather join the SpellLinks e.g. "any 4 if A / B/ C" or comma separated, but I couldn't work it out.